src/Chainweb/Miner/Coordinator.hs

```diff
@@ -79,15 +80,15 @@ type Adjustments = HM.HashMap BlockHash (T2 BlockHeight HashTarget)
 
 newtype PrevBlock = PrevBlock BlockHeader
 
-powMiner
-    :: forall cas
-    . PayloadCas cas
-    => LogFunction
+mining
+    :: forall cas. PayloadCas cas
+    => (BlockHeader -> IO BlockHeader)
```

Comment:
This is the "inner loop", which can be virtually anything, and allows us to share this code between Prod and Test scenarios.

Reply:
I think this should really be independent of the `BlockHeader` type. That type is pretty high level with many internal dependencies and invariants attached. Given that (possibly external) core mining function operate on plain bytes (just injecting the nonce and computing the hash), something like the following seems more adequate:

```
newtype BlockHeaderBytes = BlockHeaderBytes MutableByteArray 
    -- this should be a pinned byte array so that pointer so it are stable

type CoreMininingFunc = BlockHeaderBytes -> IO BlockHeaderBytes
```

Reply:
This'll be addressed elsewhere.